packages/angular/src/generators/utils/add-vitest.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  addDependenciesToPackageJson,
+  ensurePackage,
+  joinPathFragments,
+  readJson,
+  type Tree,
+} from '@nx/devkit';
+import { analogVitestAngular, nxVersion } from '../../utils/versions';
+
+export type AddVitestOptions = {
+  name: string;
+  projectRoot: string;
+  skipPackageJson: boolean;
+  strict: boolean;
+};
+
+export async function addVitest(
+  tree: Tree,
+  options: AddVitestOptions
+): Promise<void> {
+  if (!options.skipPackageJson) {
+    process.env.npm_config_legacy_peer_deps ??= 'true';
+
+    addDependenciesToPackageJson(
+      tree,
+      {},
+      {
+        '@analogjs/vitest-angular': analogVitestAngular,
+        '@analogjs/vite-plugin-angular': analogVitestAngular,
+      },
+      undefined,
+      true
+    );
+  }
+
+  const { createOrEditViteConfig, viteConfigurationGenerator } = ensurePackage<
+    typeof import('@nx/vite')
+  >('@nx/vite', nxVersion);
+
+  const relativeTestSetupPath = joinPathFragments('src', 'test-setup.ts');
+  const isEsmImplicit = readJson(tree, 'package.json').type === 'module';
```

Comment:
In Nx workspaces, applications and libraries may have their own package.json files which can define `type: module`. 

IIRC, Vite will always look for the closest `package.json` file to determine module format. 

I think this needs to be updated to the package.json path for the application/library, and if it doesn't exist, then fallback to root of workspace.

Reply:
Wouldn't it be safer to stick to vite's behavior and crawl for `package.json` parent per parent until reaching the root of the workspace?
By the way, this probably deserves a util in `@nx/devkit`. It is common for plugin authors to crawl for a file until the root of the workspace.

Reply:
Yes to both, feel free to add one :)

Reply:
Done!